src/engines/html5/media-source/adapters/native-adapter.js

```diff
@@ -977,7 +977,10 @@ export default class NativeAdapter extends BaseMediaSourceAdapter {
     const pkTextTracks = this._getPKTextTracks();
     const pkOffTrack = pkTextTracks.find(track => track.language === 'off');
     const getActiveVidTextTrackIndex = () => {
-      return this._nativeTextTracksMap.findIndex(textTrack => textTrack && this._getDisplayTextTrackModeString() === textTrack.mode);
+      const activeTrack: TextTrack = Object.entries(this._nativeTextTracksMap).find(
+        ([, textTrack]) => this._getDisplayTextTrackModeString() === textTrack.mode
+      );
+      return activeTrack ? activeTrack.index : -1;
```

Comment:
there is no index on a native text track. you should use the index (key) from the map